### PR TITLE
Feat/manager devise

### PR DESCRIPTION
**Pull request by yangalli**
# Description
- Introducing a new feature to authenticate users by token.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] It is urgent
- [ ] <span style="color:#d44037">Dangerous change</span>

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -221,6 +221,13 @@
   # Defines which key will be used when recovering the password for an account
   # config.reset_password_keys = [:email]
 
+  OmniAuth.config.logger = Rails.logger
+
+  Rails.application.config.middleware.use OmniAuth::Builder do
+    provider :google_oauth2, '333733292312-9q828nskvs3puhkqj735rgokpipgnrrb.apps.googleusercontent.com',
```
Comment (NickNish09) [inline config/initializers/devise.rb:227]: shouldn`t this be on an env variable? This seems to be some client / key thing from google

Comment (NickNish09) [reply config/initializers/devise.rb:227]: If so, we could put it on the `env.yml`

File: config/initializers/devise_token_auth.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
```
Comment (NickNish09) [inline config/initializers/devise_token_auth.rb:2]: There is a thing that Jorge used to recommend which is delete all this comments from the standard file.
What do you think?

File: spec/models/user_manager_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+RSpec.describe UserManager, type: :model do
+  describe '#create_user_for_google' do
+    let(:data) do
+      OpenStruct.new(email: 'user1@ribon.io', provider: 'google_oauth2')
+    end
+
+    it 'creates the UserManager in google with the correct params' do
+      expect { described_class.create_user_for_google(data) }.to change(described_class, :count).by(1)
+    end
```
Comment (NickNish09) [inline spec/models/user_manager_spec.rb:35]: I think that here would be nice to add a test checking if the `email` (and other possible fields) are being set correctly.
What do you think?

File: app/commands/manager/set_user_manager_tokens.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    attr_reader :id_token
+
+    def initialize(id_token:)
+      @id_token = id_token
+    end
+
+    def call
+      with_exception_handle do
+        url = "https://www.googleapis.com/oauth2/v3/tokeninfo?id_token=#{id_token}"
```
Comment (NickNish09) [inline app/commands/manager/set_user_manager_tokens.rb:15]: This is just something that we use on other files, but I would put this url in an private method, just to clean up a little bit the `call` block.
What do you think?

File: config/initializers/devise.rb (hunk 1 above)
Comment (yangalli) [reply config/initializers/devise.rb:227]: Yes, for sure! I forgot

File: config/initializers/devise_token_auth.rb (hunk 2 above)
Comment (yangalli) [reply config/initializers/devise_token_auth.rb:2]: Ok, nice!

File: app/commands/manager/set_user_manager_tokens.rb (hunk 4 above)
Comment (yangalli) [reply app/commands/manager/set_user_manager_tokens.rb:15]: Yes, good idea

File: spec/models/user_manager_spec.rb (hunk 3 above)
Comment (yangalli) [reply spec/models/user_manager_spec.rb:35]: Done!

**Review by leodhb — APPROVED** [review]

**Review by NickNish09 — APPROVED** [review]

**Pull request merged**